src/core/lib/channel/channelz_registry.cc

```diff
@@ -38,14 +38,21 @@ namespace {
 
 // singleton instance of the registry.
 ChannelzRegistry* g_channelz_registry = nullptr;
+bool g_channelz_registry_running = false;
```

Comment:
I don't think this variable is needed.  Instead, we could simply reset `g_channelz_registry` to null in `Shutdown()`.